### ISSUE TITLE
Create a "barnard59-curl" package

### ISSUE THREAD

**Issue by tpluscode** [zazuko/barnard59#98]
As proposed by @ktk, the first 100% reusable pipeline could be a self-contained replacement for curl.

Inputs: 
- a glob of local files to upload
- SPARQL endpoint details
- (optional) HTTP method

Output: none - the pipeline will use the graph store to upload

Might consider an option to silently continue (with a warning logged) when a file is not RDF, such as if a globbed directory contains a `readme.md` etc

**Comment by ktk** [comment]
Basically a version of this that accepts [globs](https://www.npmjs.com/package/glob) & chooses the correct parser per filetype:

```turtle
@base         <urn:pipeline:xrm> .
@prefix :     <https://pipeline.described.at/> .
@prefix code: <https://code.described.at/> .
@prefix rdf:  <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .

<#defaultVars>
  :variable [ a :Variable;
    :name "outputFile";
    :value "target/all.nq"
  ]
  .

<#PostToStore> a :Pipeline;
  :variables <#defaultVars>;
  :steps [
    :stepList (
      <#readFile>
      <#parseN3>
      <#setGraph>
      <#postToGraphStore>
    )
  ].

<#PutToStore> a :Pipeline;
  :variables <#defaultVars>;
  :steps [
    :stepList (
      <#readFile>
      <#parseN3>
      <#setGraph>
      <#putToGraphStore>
    )
  ].

<#readFile> a :Step;
  code:implementedBy [ a code:EcmaScript;
    code:link <node:fs#createReadStream>
  ];
  code:arguments ("outputFile"^^:VariableName).


<#parseN3> a :Step;
  code:implementedBy [ a code:EcmaScriptModule;
    code:link <node:barnard59-formats/n3.js#parse>
  ].

<#putToGraphStore>
  a                   :Step;
  code:implementedBy  [ a         code:EcmaScriptModule;
                        code:link <node:barnard59-graph-store/put.js#default> ] ;
  code:arguments      [ code:name   "endpoint" ;
                        code:value  "ENDPOINT"^^:VariableName ] ,
                      [ code:name   "user" ;
                        code:value  "USER"^^:VariableName ] ,
                      [ code:name   "password";
                        code:value  "PASSWORD"^^:VariableName ].


<#postToGraphStore>
  a                   :Step;
  code:implementedBy  [ a         code:EcmaScriptModule;
                        code:link <node:barnard59-graph-store/post.js#default> ] ;
  code:arguments      [ code:name   "endpoint" ;
                        code:value  "ENDPOINT"^^:VariableName ] ,
                      [ code:name   "user" ;
                        code:value  "USER"^^:VariableName ] ,
                      [ code:name   "password";
                        code:value  "PASSWORD"^^:VariableName ].

<#setGraph> a :Step ;
    code:implementedBy  [ a         code:EcmaScriptModule;
                        code:link <node:barnard59-rdf/setGraph.js#default> ] ;
    code:arguments      ( "https://lindas.admin.ch/elcom/electricityprice"^^code:EcmaScriptTemplateLiteral) .


```

I thought I had a version with glob somewhere but can't find it right now.

**Comment by tpluscode** [comment]
> chooses the correct parser per filetype

Related to that, I'd also make that into a step of its own and use here. In `barnard59-rdf` probably

```turtle
[
  a :Step ;
  code:implementedBy [
    a code:EcmaScriptModule ;
    code:link <node:barnard59-rdf/fs.js#parse>
  ] ;
] .
```

Removes the need to manually connect up `createReadStream` and a parser. The input chunk is a path and the output will stream its contents

**Comment by tpluscode** [comment]
Closed by #152
